woocommerce/woocommerce
# [dev] Close the CFE and PRR if the base ref of the PR doesn't originate from release branch

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This change checks for the base ref of a PR and closes it if the PR doesn't originate from the release branch.

1. Eliminates the need to manually enter the target release.
2. It also sends a failure notification on slack if the PR base if other than release.
3. In case PR details are not found due to URL being incorrect, it closes the CFE with a comment.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Follow [this guide](https://woocorerelease.wordpress.com/2024/09/18/the-new-cfe-workflow-the-complete-picture/) to setup the environment.
2. Raise a PR against `trunk`.
4. Now create a new CFE request with the PR you just raised.
5. Observe the message on your test slack channel.
6. Make sure the CFE issue gets a comment.
7. Make sure the CFE issue is closed.
8. Make sure the label is removed.
9. Now get a PR link from outside your repo.
10. Create a CFE request using that link.
11. Make sure the CFE issue is closed.
12. Make sure the CFE label is removed.
13. This time there is no slack message.
14. Now create a CFE with a PR raised against release branch.
15. Everything should work as expected.
16. Verify the slack message, github comment.
17. Repeat the steps with PRR request.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
